### ISSUE TITLE
python3 decode error for bytes

### ISSUE THREAD

**Issue by wenhuang1003** [cannatag/ldap3#364]
See #363  for detail

**Comment by cannatag** [comment]
Hi, can you give me a snippet of code to replicate the error?

Thanks,
Giovanni

**Comment by wenhuang1003** [comment]
I'm using python 3.6.1 and windows active drectory, the code is like:

```python
import  ldap3

server = ldap3.Server([server], use_ssl=True, get_info=ldap3.ALL)
ldap_client = ldap3.Connection(
        server,
        user=[username],
        password=[password],
        authentication=ldap3.NTLM,
        auto_bind=True
        )
ldap_filter = '(objectGUID=[objectGUID])'
ldap_client.search([BASE_DN], ldap_filter,
                             attributes=ldap3.ALL_ATTRIBUTES)
print(g_ldap_client.entries)
```

replace `[param]` with yours

**Comment by cannatag** [comment]
Hi,
I've fixed the PyAs1 error. Now you can search for GUID with code similar to:

```
# substitute the <MY_*> variables with the correct values for your environment
SERVER = '<MY_SERVER>'
USERNAME = '<MY_USER>'  # in the format DOMAIN\\USERNAME (remember the double "\")
PASSWORD = '<MY_PASSWORD>'
GUID_str = '<MY_GUID_STR> # (in the format 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx)
GUID = validate_uuid_le(GUID_str)
BASE_DN = '<MY_USER_BASE>'
server = ldap3.Server(SERVER, use_ssl=True, get_info=ldap3.ALL)
ldap_client = ldap3.Connection(
    server,
    user=USERNAME,
    password=PASSWORD,
    authentication=ldap3.NTLM,
    auto_bind=True
)

ldap_filter = '(objectGUID=' + escape_bytes(GUID) + ')'
ldap_client.search(BASE_DN, ldap_filter, attributes=ldap3.ALL_ATTRIBUTES)
print(ldap_client.entries)

```

If your GUID_str is correct you should get the user object back:

```
[DN: CN=Giovanni,CN=Users,DC=AD2012,DC=LAB - STATUS: Read - READ TIME: 2017-06-25T00:28:10.546841
    accountExpires: 1601-01-01 00:00:00+00:00
    adminCount: 1
    badPasswordTime: 2017-06-24 14:57:09.856527+00:00
    badPwdCount: 0
    cn: Giovanni
    codePage: 0
    countryCode: 0
    dSCorePropagationData: 2014-10-07 18:54:15+00:00
                           2014-10-07 18:54:14+00:00
                           2014-10-07 18:39:00+00:00
                           1601-01-01 18:12:16+00:00
    description: Built-in account for administering the computer/domain
    distinguishedName: CN=Giovanni,CN=Users,DC=AD2012,DC=LAB
    instanceType: 4
    isCriticalSystemObject: True
    lastLogoff: 1601-01-01 00:00:00+00:00
    lastLogon: 2017-06-24 15:03:17.122400+00:00
    lastLogonTimestamp: 2017-06-24 14:59:16.937778+00:00
    logonCount: 57
    logonHours: b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff'
    memberOf: CN=Group Policy Creator Owners,CN=Users,DC=AD2012,DC=LAB
              CN=Domain Admins,CN=Users,DC=AD2012,DC=LAB
              CN=Enterprise Admins,CN=Users,DC=AD2012,DC=LAB
              CN=Schema Admins,CN=Users,DC=AD2012,DC=LAB
              CN=Administrators,CN=Builtin,DC=AD2012,DC=LAB
    name: Giovanni
    objectCategory: CN=Person,CN=Schema,CN=Configuration,DC=AD2012,DC=LAB
    objectClass: top
                 person
                 organizationalPerson
                 user
    objectGUID: 73bd0474-6457-41d4-8208-ccc360759c54
    objectSid: S-1-5-21-3365175042-1191635722-789601614-500
    primaryGroupID: 513
    pwdLastSet: 2017-06-24 14:57:16.648466+00:00
    sAMAccountName: Giovanni
    sAMAccountType: 805306368
    uSNChanged: 3454301
    uSNCreated: 8196
    userAccountControl: 512
    whenChanged: 2017-06-24 14:59:16+00:00
    whenCreated: 2014-10-07 18:34:50+00:00
```

You must get the latest code in dev for this to work. Let me know if it's ok for you.

Bye,
Giovanni

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.

**Comment by wenhuang1003** [comment]
Thanks a lot!  I don't need to edit the source after  `pip install ` now!
